src/TimeModeling/TimeModeling.jl

```diff
@@ -73,5 +55,28 @@ include("seismic_preconditioners.jl")
 #############################################################################
 # Utility types
 const SourceTypes = Union{judiVector, Tuple{judiWeights, judiLRWF}}
+# PDE types are callable w.r.t non-linear parameters, i.e F(model) or F(;m=m, epsilon=new_epsilon)
+const pde_types = Union{judiModeling, judiModelingAdjoint, judiPDEfull, judiPDE, judiPDEadjoint,
+                        judiJacobian, judiJacobianExQ, judiPDEextended}
+
+
+if VERSION>v"1.2"
+  function (F::pde_types)(m::Model)
+    Fl = deepcopy(F)
+    Fl.model.n = m.n
+    Fl.model.d = m.d
+    Fl.model.o = m.o
+    for (k, v) in m.params
+      Fl.model.params[k] = v
+    end
+    Fl
```

Comment:
Why not

`Fl.model = deepcopy(m)`

Reply:
No, `Fl` is immutable so can't modify its attribute, only `Fl.m` is mutable and can be modified.